src/zypper_plugins/tracer_upload.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+#
+# Copyright 2019 ATIX AG
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+
+import json
+import httplib
+from os import path, environ
+import sys
+import logging
+from rhsm.config import RhsmConfigParser, initConfig
+from zypp_plugin import Plugin
+
+sys.path.append('/usr/share/rhsm')
+from subscription_manager.identity import ConsumerIdentity
+
+class TracerUploadPlugin(Plugin):
+    def __init__(self):
+        Plugin.__init__(self)
+        self.num1 = self.num2 = None
+        self.description = "Upload traces to katello"
+        self.cleanup = ""
+        self.userdat = {}
+
+    def collect_data(self):
+        """
+        Return a array with nested arrays
+        containing name, how to restart & app type
+        for every package that needs restarting
+        """
+
+        apps = {}
+
+        reboot_file = "/var/run/reboot-needed"
```

Comment:
Perhaps this is a good constant? I'm also wondering how you chose this filename. Debian uses `/var/run/reboot-required` for this.

Reply:
See https://github.com/openSUSE/zypper/blob/e2dd8806672f852e3a7a41078054ee71c467dc6e/src/commands/needs-rebooting.cc#L31